DEVELOPERS_GUIDE.md

```diff
@@ -0,0 +1,119 @@
+# Developers Guide
+The contents fo this document are meant to help a developer understand the system requirements, coding guidelines, and operation of the CMR needed to work on it as an active contributer.
+
+
+**Table of content:**
+ - [Technical Stack](#technical-stack)
+ - [Running CMR](#running-the-cmr)
+
+
+## Technical Stack
+The CMR utilizes the following technologes to work:
+- [Java](https://docs.oracle.com/en/java/)
+- [Clojure](https://clojure.org)
+- [Lieningen](https://leiningen.org)
+- [Docker](https://www.docker.com)
+- [Elastic](https://www.elastic.co)
+- [Oracle](https://www.oracle.com/database)
+
+
+## Clojure & Leiningen
+
+### Clojure
+
+#### What is Clojure?
+Clojure is a dynamic development environment where you interact with your program while you write it, growing and adding to it while it’s running. To work with Clojure you need an editor that supports evaluation in source files and structural editing (working with nested forms in addition to character editing).
+
+#### Why we use it
+Clojure meets its goals by: embracing an industry-standard, open platform - the JVM; modernizing a venerable language - Lisp; fostering functional programming with immutable persistent data structures; and providing built-in concurrency support via software transactional memory and asynchronous agents. The result is robust, practical, and fast.
+
+### Leiningen
+
+#### What is Leiningen?
+Leiningen is a build automation and dependency management tool for the simple configuration of software projects written in the Clojure programming language.
+
+#### Why we use it
+Leiningen is the easiest way to use Clojure. With a focus on project automation and declarative configuration, it gets out of your way and lets you focus on your code.
+
+
+
+## Running the CMR
+The CMR is a system consisting of many services. The services can run
+individually or in a single process. Running in a single process makes
+local development easier because it avoids having to start many different
+processes. The `dev-system` project allows the CMR to run from a single REPL
+or JAR file. If you are developing a client against the CMR you can build and
+run the entire CMR with no external dependencies from this JAR file and use
+that instance for local testing. The sections below contain instructions for
+running the CMR as a single process or as many processes.
+
+### Using the `cmr` CLI Tool
```

Comment:
is there a plan to include any information on using Visual Studio Code or emacs?